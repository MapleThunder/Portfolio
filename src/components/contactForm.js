import React from "react";
import styled from "styled-components";
import { graphql, useStaticQuery } from "gatsby";
import GatsbyImage from "gatsby-image";

const Form = styled.form`
  input,
  textarea {
    max-width: 500px;
    width: 100%;
  }

  > div + div {
    margin-top: 10px;
  }

  button[type="submit"] {
    margin-top: 25px;
    display: flex;
    align-items: center;

    .envelope {
      margin-left: 5px;
    }
  }

  @media (min-width: 700px) {
    input {
      width: 45%;
    }
    textarea {
      width: 65%;
    }
  }
  @media (min-width: 900px) {
    input {
      width: 25%;
    }
    textarea {
      width: 33%;
    }
  }
`;

const ContactForm = () => {
  const data = useStaticQuery(graphql`
    {
      file(relativePath: { eq: "envelope.png" }) {
        childImageSharp {
          fixed(width: 20, height: 20) {
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
    }
  `);

  return (
    <section id="contact">
      <h2 id="contact">Contact Me</h2>
      <Form
        method="post"
        netlify-honeypot="bot-field"
        data-netlify="true"
        name="contact"
      >
        <input type="hidden" name="bot-field" />
        <input type="hidden" name="form-name" value="contact" />

        <div>
          <label for="email">Email</label>
          <input type="email" name="email" id="email" />
        </div>
        <div>
          <label for="name">Name</label>
          <input type="text" name="name" id="name" />
        </div>
        <div>
          <label for="message">Message</label>
          <textarea
            name="message"
            id="message"
            placeholder="Enter your message."
          ></textarea>
        </div>
        <button type="submit">
          Send
          <GatsbyImage
            alt="Envelope"
            className="envelope"
            {...data.file.childImageSharp}
          />
        </button>
      </Form>
    </section>
  );
};

export default ContactForm;
