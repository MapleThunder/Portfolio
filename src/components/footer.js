import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { COLOURS } from "../styles";

const FooterWrapper = styled.footer`
  background: ${COLOURS.bdazzledBlue};
  height: 100px;
  width: 100%;
  bottom: 0;
`;

const Footer = ({ siteTitle }) => {
  return <footer></footer>;
};

Footer.propTypes = {
  siteTitle: PropTypes.string,
};

Footer.defaultProps = {
  siteTitle: ``,
};

export default Footer;
