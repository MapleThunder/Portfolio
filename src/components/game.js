import React from "react";
import Unity, { UnityContext } from "react-unity-webgl";

const unityContext = new UnityContext({
  loaderUrl: "/ProjectBoost/Build/ProjectBoost.loader.js",
  dataUrl: "/src/ProjectBoost/Build/ProjectBoost.data",
  frameworkUrl: "/src/ProjectBoost/Build/ProjectBoost.framework.js",
  codeUrl: "/src/ProjectBoost/Build/ProjectBoost.wasm",
});

export const Game = () => {
  return (
    <div>
      GAME
      {/* <Unity unityContext={unityContext} /> */}
    </div>
  );
};
