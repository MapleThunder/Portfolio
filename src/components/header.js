import { Link } from "gatsby";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { COLOURS } from "./styles";
import { ThemeToggle } from "./themeToggle";

const HeaderWrapper = styled.header`
  background: ${COLOURS.main};
`;

const HeaderContainer = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 1rem;
  padding-left: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;

  h1 {
    margin: 0;
  }
  a {
    color: white;
    text-decoration: none;
  }
`;

const Header = ({ siteTitle }) => (
  <HeaderWrapper>
    <HeaderContainer>
      <h1>
        <Link to="/">{siteTitle}</Link>
      </h1>
      <ThemeToggle />
    </HeaderContainer>
  </HeaderWrapper>
);

Header.propTypes = {
  siteTitle: PropTypes.string,
};

Header.defaultProps = {
  siteTitle: ``,
};

export default Header;
