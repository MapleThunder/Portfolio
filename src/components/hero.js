import React from "react";
import styled from "styled-components";
import { graphql, useStaticQuery } from "gatsby";
import Img from "gatsby-image";

const StyledSection = styled.section`
  display: block;
  padding: 0;

  .hero-text {
    .first-line {
      font-size: 2rem;
      padding-bottom: 25px;
    }
  }

  .hero-image {
    width: 100%;
    justify-self: end;
    .me {
      border: solid 1px var(--grey);
      border-radius: 50%;
      box-shadow: var(--level-2);
    }
  }
  .title {
    margin-bottom: 0;
  }
  .subtitle {
    margin-top: 0;
  }
  .highlighted {
    box-shadow: inset 0 -2.5rem 0 var(--redSalsa);
    font-size: inherit;
  }

  @media only screen and (min-width: 600px) {
    display: flex;
    align-items: center;
    justify-content: space-between;

    & > div {
      width: 75%;
    }
    .hero-image {
      width: 25%;
    }
  }
  @media only screen and (min-width: 800px) {
    & > div {
      width: 48%;
    }
    .hero-image {
      width: 40%;
    }
  }
`;

const Hero = () => {
  const data = useStaticQuery(graphql`
    query {
      file(relativePath: { eq: "me.jpg" }) {
        childImageSharp {
          fluid(maxWidth: 2100) {
            base64
            tracedSVG
            aspectRatio
            src
            srcSet
            srcWebp
            srcSetWebp
            sizes
            originalImg
            originalName
          }
        }
      }
      hero: allMarkdownRemark(
        filter: { frontmatter: { type_id: { eq: "hero-front" } } }
      ) {
        edges {
          node {
            frontmatter {
              emoji
              greetings
              subtitleHighlight
              subtitlePrefix
              title
            }
            html
          }
        }
      }
    }
  `);
  const { frontmatter, html } = data.hero.edges[0].node;

  return (
    <StyledSection id="hero">
      <div className="hero-text">
        <h1 data-testid="hero-title" className="title">
          {frontmatter.greetings}{" "}
          <span role="img" aria-label="emoji" style={{ fontSize: "inherit" }}>
            {frontmatter.emoji}
          </span>
          <br />
          {frontmatter.title}
        </h1>
        <h2 className="subtitle">
          {frontmatter.subtitlePrefix}{" "}
          <span className="highlighted">{frontmatter.subtitleHighlight}</span>.
        </h2>
        <div
          className="description"
          dangerouslySetInnerHTML={{ __html: html }}
        ></div>
      </div>
      <div className="hero-image">
        <Img
          className="me"
          alt="A picture of me at the beach in SÃ£o Jacinto, Portugal."
          {...data.file.childImageSharp}
        />
      </div>
    </StyledSection>
  );
};

export default Hero;
