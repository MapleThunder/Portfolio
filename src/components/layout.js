/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.com/docs/use-static-query/
 */

import React from "react";
import PropTypes from "prop-types";
import { useStaticQuery, graphql } from "gatsby";
import styled from "styled-components";

import Header from "./header";
import "./styles/layout.css";

const LayoutWrapper = styled.div`
  background-color: var(--bg);
  color: var(--textNormal);
  transition: color 0.2s ease-out, background 0.2s ease-out;
`;
const MainLayout = styled.main`
  /* margin: 1rem auto; */
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 40px;
`;

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <LayoutWrapper>
      <Header siteTitle={data.site.siteMetadata?.title || `Title`} />
      <MainLayout>{children}</MainLayout>
    </LayoutWrapper>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
