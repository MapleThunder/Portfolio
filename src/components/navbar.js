import React from "react";
import styled from "styled-components";
import { Link } from "gatsby";

import Burger from "./burger";

const Nav = styled.nav`
  width: 100%;
  height: 55px;
  padding: 0;
  display: flex;
  justify-content: space-between;
  a.logo {
    font-size: var(--h2);
    font-family: var(--headingFont);
    color: white;
    text-decoration: none;
    line-height: 2.5rem;
    border: solid 2px transparent;

    &:hover,
    &:focus {
      background-color: var(--accent);
      border: solid 2px var(--accent);
      color: var(--textColour);
    }
  }
`;

const Navbar = () => {
  return (
    <Nav>
      <Link to="/" className="logo">
        NB
      </Link>
      <Burger />
    </Nav>
  );
};

export default Navbar;
