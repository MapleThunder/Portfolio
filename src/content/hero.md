---
greetings: "Ola"
emoji: "ðŸ‘‹"
title: "I'm Niko Bentley"
subtitlePrefix: "I build and design "
subtitleHighlight: "things for the web"
type_id: "hero-front"
---

Web Developer with a passion for learning.  
Based in Halifax, Nova Scotia.
