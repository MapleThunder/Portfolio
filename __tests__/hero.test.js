import React from "react";
import { render, cleanup } from "@testing-library/react";
import { PureHero as Hero } from "../src/components/hero";

beforeEach(cleanup); // clean the DOM !

describe("Hero", () =>
  it("renders correctly", () => {
    const node = {
      frontmatter: {
        emoji: "ðŸ‘‹",
        greetings: "Ola",
        subtitleHighlight: "things for the web",
        subtitlePrefix: "I build and design ",
        title: "I'm Niko Bentley",
      },
      html:
        "<p>Web Developer with a passion for learning.<br>\nBased in Halifax, Nova Scotia.</p>",
    };
    const image = {
      childImageSharp: {
        fluid: {
          src: "/static/c392e3865190c633b0435677eecb6f43/6ed45/me.jpg",
          srcWebp: "/static/c392e3865190c633b0435677eecb6f43/90b9d/me.webp",
          srcSetWebp:
            "/static/c392e3865190c633b0435677eecb6f43/f5db0/me.webp 525w,\n/static/c392e3865190c633b0435677eecb6f43/565e9/me.webp 1050w,\n/static/c392e3865190c633b0435677eecb6f43/90b9d/me.webp 2100w",
          sizes: "(max-width: 2100px) 100vw, 2100px",
        },
      },
    };
    const { getByTestId, asFragment } = render(
      <Hero node={node} image={image} />
    );

    expect(getByTestId("hero-title")).toHaveTextContent("I'm Niko Bentley");
    expect(asFragment(<Hero node={node} image={image} />)).toMatchSnapshot();
  }));
