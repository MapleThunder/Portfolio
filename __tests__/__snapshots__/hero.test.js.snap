// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<DocumentFragment>
  <section
    class="sc-bdfBwQ iAWJTK"
    id="hero"
  >
    <div
      class="hero-text"
    >
      <h1
        class="title"
        data-testid="hero-title"
      >
        Ola 
        <span
          aria-label="emoji"
          role="img"
        >
          ðŸ‘‹
        </span>
        <br />
        I'm Niko Bentley
      </h1>
      <h2
        class="subtitle"
      >
        I build and design  
        <span
          class="highlighted"
        >
          things for the web
        </span>
        .
      </h2>
      <div
        class="description"
      >
        <p>
          Web Developer with a passion for learning.
          <br />
          
Based in Halifax, Nova Scotia.
        </p>
      </div>
    </div>
    <div
      class="hero-image"
    >
      <div
        class="me gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="width: 100%;"
        />
        <picture>
          <source
            sizes="(max-width: 2100px) 100vw, 2100px"
            srcset="/static/c392e3865190c633b0435677eecb6f43/f5db0/me.webp 525w,
/static/c392e3865190c633b0435677eecb6f43/565e9/me.webp 1050w,
/static/c392e3865190c633b0435677eecb6f43/90b9d/me.webp 2100w"
            type="image/webp"
          />
          <img
            alt="A picture of me at the beach in SÃ£o Jacinto, Portugal."
            loading="lazy"
            sizes="(max-width: 2100px) 100vw, 2100px"
            src="/static/c392e3865190c633b0435677eecb6f43/6ed45/me.jpg"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source type='image/webp' srcset="/static/c392e3865190c633b0435677eecb6f43/f5db0/me.webp 525w,
/static/c392e3865190c633b0435677eecb6f43/565e9/me.webp 1050w,
/static/c392e3865190c633b0435677eecb6f43/90b9d/me.webp 2100w" sizes="(max-width: 2100px) 100vw, 2100px" /&gt;&lt;source srcset="undefined" sizes="(max-width: 2100px) 100vw, 2100px" /&gt;&lt;img loading="lazy" sizes="(max-width: 2100px) 100vw, 2100px" src="/static/c392e3865190c633b0435677eecb6f43/6ed45/me.jpg" alt="A picture of me at the beach in SÃ£o Jacinto, Portugal." style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
